pkg/injector/envoy_container.go

```diff
@@ -40,7 +40,7 @@ func getPlatformSpecificSpecComponents(meshConfig v1alpha2.MeshConfig, podOS str
 
 func getEnvoySidecarContainerSpec(pod *corev1.Pod, meshConfig v1alpha2.MeshConfig, originalHealthProbes models.HealthProbes, podOS string) corev1.Container {
 	// cluster ID will be used as an identifier to the tracing sink
-	clusterID := fmt.Sprintf("%s.%s", pod.Spec.ServiceAccountName, pod.Namespace)
+	clusterID := fmt.Sprintf("%s.%s", pod.Spec.ServiceAccountName, pod.ObjectMeta.Namespace)
```

Comment:
The problem here is that `pod.Namespace` is unset in the API request to the webhook, so you can't rely on it. You instead need to make use of the `namespace` variable derived from `req.Namespace` in `createPatch()` by passing it to `getEnvoySidecarContainerSpec()`.

Reply:
I never knew this! Do you have a link to some docs or code that shows namespace being removed?

Reply:
This is the reason we rely on the following code:
https://github.com/openservicemesh/osm/blob/a551827f40c4ff858d1ebb938db1f6a680abe950/pkg/injector/patch.go#L28

I strongly suspect that might be the underlying cause here, as it was definitely an issue I ran into when I implemented automatic sidecar injection a couple of years ago.

Reply:
@whitneygriffith does this make sense?

Reply:
@shashankram @keithmattix this makes sense! Thanks!

Reply:
Why do we unset the namespace? @shashankram

Reply:
We do not unset the namespace, but rather the namespace is unset in the request body from the API server. This is what I observed a long time ago and the reason why we use `req.Namespace` and not `pod.Namespace` for the namespace value.

Reply:
Relying on `req.namespace` resolved the missing namespace in AKS environment